torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -174,6 +174,18 @@ static Value* tryMatchArgument(
       auto& ostream = err()
           << arg.formatTypeMismatchMsg(value->type()->python_str());
 
+      if (auto pt = value->type()->cast<TensorType>()) {
+        if (pt->kind() == TypeKind::TensorType && pt->isInferredType()) {
+          std::string inferred_type_hint;
+          inferred_type_hint = c10::str(
+              "Inferred the value for argument '",
+              arg.name(),
+              "' to be of type 'Tensor' ",
+              "because it was not annotated with an explicit type.\n");
+          ostream << inferred_type_hint;
```

Comment:
This looks like it was copied from `function_schema.h`; can we refactor this somehow to avoid duplicating code?